crates/test_utils/src/mark.rs

```diff
@@ -0,0 +1,78 @@
+//! This module implements manually tracked test coverage, which is useful for
+//! quickly finding a test responsible for testing a particular bit of code.
+//!
+//! See <https://matklad.github.io/2018/06/18/a-trick-for-test-maintenance.html>
+//! for details, but the TL;DR is that you write your test as
+//!
+//! ```
+//! #[test]
+//! fn test_foo() {
+//!     mark::check!(test_foo);
+//! }
+//! ```
+//!
+//! and in the code under test you write
+//!
+//! ```
+//! # use test_utils::mark;
+//! # fn some_condition() -> bool { true }
+//! fn foo() {
+//!     if some_condition() {
+//!         mark::hit!(test_foo);
+//!     }
+//! }
+//! ```
+//!
+//! This module then checks that executing the test indeed covers the specified
+//! function. This is useful if you come back to the `foo` function ten years
+//! later and wonder where the test are: now you can grep for `test_foo`.
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[macro_export]
+macro_rules! _hit {
+    ($ident:ident) => {{
+        #[cfg(test)]
+        {
+            extern "C" {
+                #[no_mangle]
+                static $ident: std::sync::atomic::AtomicUsize;
```

Comment:
can we add some prefix to the name, in case someone does `mark::hit!(main)` or something like that?

Reply:
I think not without proc_macros, as we'd either need `concat_idents`, which doesn't work, or `#[link_name = stringify!(prefix $ident)]` which doesn't work either.

Reply:
And yeah, `hit!(main)` segfaults :-( 

I guess that's fine for `#[cfg(test)]` code.

Reply:
How about using `line! + file! + col!` macro with `stringify` similar to common C macro tricks ?